python/lsst/ctrl/oods/butlerAttendant.py

```diff
@@ -334,6 +273,57 @@ def clean(self):
                 try:
                     uri.remove()
                 except Exception as e:
-                    LOGGER.info("error removing %s: %s", uri, e)
+                    LOGGER.warning("couldn't remove %s: %s", uri, e)
 
         butler.pruneDatasets(ref, purge=True, unstore=True)
+
+    def rawexposure_info(self, data):
+        """Return a sparsely initialized dictionary
+
+        Parameters
+        ----------
+        data: `RawFileData`
+            information about the raw file
+
+        Returns
+        -------
+        info: `dict`
+            Dictionary with file name and dataId elements
+        """
+        info = dict()
+        dataset = data.datasets[0]
+        info["FILENAME"] = "??"
+        dataId = dataset.dataId
+        info["CAMERA"] = dataId.get("instrument", "??")
+        info["OBSID"] = dataId.get("exposure", "??")
+        info["RAFT"] = self.extract_info_val(dataId, "raft", "R")
+        info["SENSOR"] = self.extract_info_val(dataId, "detector", "S")
+        return info
+
+    def undef_metadata(self, filename):
+        """Return a sparsely initialized metadata dictionary
+
+        Parameters
+        ----------
+        filename: `str`
+            name of the file specified by ingest
+
+        Returns
+        -------
+        info: `dict`
+            Dictionary containing file name and placeholders
+        """
+        info = dict()
+        info["FILENAME"] = filename
+        info["CAMERA"] = "UNDEF"
+        info["OBSID"] = "??"
+        info["RAFT"] = "R??"
+        info["SENSOR"] = "S??"
+        return info
+
+    def definer_run(self, file_datasets):
+        for fds in file_datasets:
+            refs = fds.refs
+            ids = [ref.dataId for ref in refs]
+            self.visit_definer.run(ids)
```

Comment:
If there's a failure, the following datasets don't get visits defined?  You may need a `try:` block here.

Reply:
Added